lib/wasi/src/syscalls/mod.rs

```diff
@@ -2512,10 +2498,15 @@ pub fn poll_oneoff(
     __WASI_ESUCCESS
 }
 
+// pub fn proc_exit(env: &WasiEnv, code: __wasi_exitcode_t) -> Result<(), WasiError> {
+//     // panic!("Exit: {}", code);
+//     Err(WasiError::Exit(code))
+//     // RuntimeError::raise(Box::new(WasiError::Exit(code)));
+// }
 pub fn proc_exit(env: &WasiEnv, code: __wasi_exitcode_t) {
-    debug!("wasi::proc_exit, {}", code);
-    RuntimeError::raise(Box::new(WasiError::Exit(code)));
-    unreachable!();
+    // panic!("Exit: {}", code);
+    // Err(WasiError::Exit(code))
+    // RuntimeError::raise(Box::new(WasiError::Exit(code)));
 }
```

Comment:
We need to clarify this.

Reply:
Fixed in https://github.com/wasmerio/wasmer/pull/2491/commits/650352ddeca4e130fcfe45312f4a9ccdd1cea841.